src/content/docs/en/guides/media/cloudinary.mdx

```diff
@@ -6,11 +6,214 @@ stub: true
+
+## Configuring Your Account
+
+Create a new `.env` file in the root of your project and add your Cloudinary credentials:
+
+```shell
+PUBLIC_CLOUDINARY_CLOUD_NAME="<Your Cloud Name>"
+
+// Only needed if using CldUploadWidget or cldAssetsLoader
+PUBLIC_CLOUDINARY_API_KEY="<Your API Key>"
+CLOUDINARY_API_SECRET="<Your API Secret>"
+```
+
+## Adding Images
+
+To add an image, you can use the CldImage component, which will automatically optimize your image and give you access to the Transformations API.
+
+```jsx
+---
+import { CldImage } from 'astro-cloudinary';
+---
+<CldImage
+  src="<Public ID>"
+  width="<Width>"
+  height="<Height>"
+  alt="<Description>"
+/>
+```
+
+[Learn more about CldImage](https://astro.cloudinary.dev/cldimage/basic-usage).
+
+## Adding Videos
+
+To add a video, you can use the CldVideoPlayer component, which will embed your video automatically optimized using the [Cloudinary Video Player](https://cloudinary.com/documentation/cloudinary_video_player).
+
+```jsx
+---
+import { CldVideoPlayer } from 'astro-cloudinary';
+---
+<CldVideoPlayer
+  src="<Public ID>"
+  width="<Width>"
+  height="<Height>"
+/>
+```
+
+[Learn more about CldVideoPlayer](https://astro.cloudinary.dev/cldvideoplayer/basic-usage).
+
+## Uploading Files
+
+To add file uploading in your website or app's UI, you can use the CldUploadWidget, which will embed the [Cloudinary Upload Widget](https://cloudinary.com/documentation/upload_widget).
+
+For unsigned uploads, create an unsigned [Upload Preset](https://cloudinary.com/documentation/upload_presets#banner) and add:
+
+```jsx
+---
+import { CldUploadWidget } from 'astro-cloudinary';
+---
+<CldUploadWidget uploadPreset="<Upload Preset>">
+  <button>Upload</button>
+</CldUploadWidget>
+```
+
+For signed uploads, you can find [a guide and example](https://astro.cloudinary.dev/clduploadwidget/signed-uploads) on the Astro Cloudinary docs.
+
+[Learn more about CldUploadWidget](https://astro.cloudinary.dev/clduploadwidget/basic-usage).
+
+## Loading Images & Videos
+
+The Astro Content Layer can be used to load Cloudinary assets with the `cldAssetsLoader`.
+
+To load images or videos, add:
+
+```js
+import { defineCollection } from 'astro:content';
+import { cldAssetsLoader } from 'astro-cloudinary/loaders';
+
+export const collections = {
+  assets: defineCollection({
+    loader: cldAssetsLoader({
+      folder: '<Folder>' // Optional, without loads root directory
+    })
+  }),
+}
+```
+
+You can then use the [getCollection or getEntry functions](/en/guides/content-collections/#querying-collections) to select one or many images or videos from your collection.
+
+[Learn more about cldAssetsLoader](https://astro.cloudinary.dev/cldassetsloader/basic-usage).
+
+## Generating Social Media Cards
+
+Social Media Cards are useful for increasing engagement on social posts.
+
+To create a Social card, generate a Cloudinary URL and utilize Open Graph and Twitter meta tags to embed the image on your website:
```

Comment:
```suggestion
The Astro Cloudinary SDK provides a `getCldOgImageUrl()` helper for generating and using URLs for your images. Use this when you need a URL instead of a component to display your image. 

One common use for a URL is for an Open Graph image in `<meta>` tags for social media cards. This helper, like the components, provides you access to Cloudinary transformations to create dynamic, unique social cards for any of your pages.

The following example shows the necessary `<meta>` tags for a social media card, using `getCldOgImageUrl()` to generate an Open Graph image:
```